apps/web/pages/api/v1/environments/[environmentId]/surveys/[surveyId]/responses/[submissionId]/index.ts

```diff
@@ -49,11 +49,18 @@ export default async function handle(req: NextApiRequest, res: NextApiResponse)
   // Delete /api/environments[environmentId]/surveys/[surveyId]/responses/[responseId]
   // Deletes a single survey
   else if (req.method === "DELETE") {
-    const submissionId = req.query.submissionId?.toString();
-    const prismaRes = await prisma.response.delete({
-      where: { id: submissionId },
+    const responseId = req.query.submissionId?.toString();
+    const response = await prisma.response.findUnique({
+      where: { id: responseId },
+      select: { displayId: true },
     });
-    return res.json(prismaRes);
+    if (response?.displayId) {
```

Comment:
We can also remove this additional logic when referencing the response in the display because the delete operation cascades down.